lib/Messenger/Handler/AssetUpdateTasksHandler.php

```diff
@@ -68,18 +67,21 @@ private function processDocument(Asset\Document $asset): void
             $asset->save(['versionNote' => 'PDF scan result']);
         }
 
+        $savedNeeded = false;
         $pageCount = $asset->getCustomSetting('document_page_count');
         if (!$pageCount || $pageCount === 'failed') {
-            if ($asset->processPageCount()) {
-                $this->saveAsset($asset);
-            }
+            $savedNeeded |= $asset->processPageCount();
```

Comment:
```suggestion
            $savedNeeded = $asset->processPageCount();
```
@cancan101 Is there a special reason why you use here a Bitwise OR assigment? Because for me that doesn't make sense with 2 booleans

Reply:
1. Avoid short circuit
2. Conciseness
3. php doesn't have `||=` (also see (1))

I can change to `$savedNeeded = $asset->processPageCount() || $savedNeeded;` or similar.

In this case the variable is only `false` above but if you had a chain of these and did still want to run the function the non short circuit is prefered.

Reply:
@cancan101 Thanks now I understand it. 
We have decided internally that this is a BC-Break and we need to move this to Pimcore 13, because if an application relies on the exception that won't work anymore with your PR